[Bug]: Fix upgrade test

### Is there an existing issue already for this bug?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [X] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [X] I have read the troubleshooting guide and I think this is a new bug.

### Contact Details

_No response_

### Version

1.22.1

### What version of Kubernetes are you using?

1.28

### What is your Kubernetes environment?

Self-managed: kind (evaluation)

### How did you install the operator?

YAML manifest

### What happened?

After we added the `--server-side` label to support 0.29.x k8s libs, we added the `--force-conflicts` only to the documentation but we forgot to added to the upgrade test, meaning that it's failing on every branch. 

### Cluster resource

```shell
kubectl apply --server-side -f ~/src/cloudnative-pg/dist/operator-manifest.yaml
namespace/cnpg-system serverside-applied
customresourcedefinition.apiextensions.k8s.io/backups.postgresql.cnpg.io serverside-applied
customresourcedefinition.apiextensions.k8s.io/clusters.postgresql.cnpg.io serverside-applied
customresourcedefinition.apiextensions.k8s.io/poolers.postgresql.cnpg.io serverside-applied
customresourcedefinition.apiextensions.k8s.io/scheduledbackups.postgresql.cnpg.io serverside-applied
serviceaccount/cnpg-manager serverside-applied
clusterrole.rbac.authorization.k8s.io/cnpg-manager serverside-applied
clusterrolebinding.rbac.authorization.k8s.io/cnpg-manager-rolebinding serverside-applied
configmap/cnpg-controller-manager-env-c9f2hmktt4 serverside-applied
configmap/cnpg-default-monitoring serverside-applied
service/cnpg-webhook-service serverside-applied
deployment.apps/cnpg-controller-manager serverside-applied
Apply failed with 3 conflicts: conflicts with "kubectl-client-side-apply" using admissionregistration.k8s.io/v1:
- .webhooks[name="mbackup.cnpg.io"].rules
- .webhooks[name="mcluster.cnpg.io"].rules
- .webhooks[name="mscheduledbackup.cnpg.io"].rules
Please review the fields above--they currently have other managers. Here
are the ways you can resolve this warning:
* If you intend to manage all of these fields, please re-run the apply
  command with the `--force-conflicts` flag.
* If you do not intend to manage all of the fields, please edit your
  manifest to remove references to the fields that should keep their
  current managers.
* You may co-own fields by updating your manifest to match the existing
  value; in this case, you'll become the manager if the other manager(s)
  stop managing the field (remove it from their configuration).
See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts
Apply failed with 4 conflicts: conflicts with "kubectl-client-side-apply" using admissionregistration.k8s.io/v1:
- .webhooks[name="vbackup.cnpg.io"].rules
- .webhooks[name="vcluster.cnpg.io"].rules
- .webhooks[name="vpooler.cnpg.io"].rules
- .webhooks[name="vscheduledbackup.cnpg.io"].rules
Please review the fields above--they currently have other managers. Here
are the ways you can resolve this warning:
* If you intend to manage all of these fields, please re-run the apply
  command with the `--force-conflicts` flag.
* If you do not intend to manage all of the fields, please edit your
  manifest to remove references to the fields that should keep their
  current managers.
* You may co-own fields by updating your manifest to match the existing
  value; in this case, you'll become the manager if the other manager(s)
  stop managing the field (remove it from their configuration).
See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts
make: *** [Makefile:183: deploy] Error 1
make: Leaving directory '/home/zeus/src/cloudn
```


### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/Makefile
+++ b/Makefile
@@ -180,7 +180,7 @@ uninstall: manifests kustomize ## Uninstall CRDs from a cluster.
 	$(KUSTOMIZE) build config/crd | kubectl delete -f -
 
 deploy: generate-manifest ## Deploy controller in the configured Kubernetes cluster in ~/.kube/config.
-	kubectl apply --server-side -f ${OPERATOR_MANIFEST_PATH}
+	kubectl apply --server-side --force-conflicts -f ${OPERATOR_MANIFEST_PATH}
 
 generate-manifest: manifests kustomize ## Generate manifest used for deployment.
 	set -e ;\

--- a/tests/e2e/upgrade_test.go
+++ b/tests/e2e/upgrade_test.go
@@ -358,7 +358,7 @@ var _ = Describe("Upgrade", Label(tests.LabelUpgrade, tests.LabelNoOpenshift), O
 		By(fmt.Sprintf("applying manager manifest %s", operatorManifestFile), func() {
 			// Upgrade to the new version
 			_, stderr, err := testsUtils.Run(
-				fmt.Sprintf("kubectl apply --server-side -f %v", operatorManifestFile))
+				fmt.Sprintf("kubectl apply --server-side --force-conflicts -f %v", operatorManifestFile))
 			Expect(err).NotTo(HaveOccurred(), "stderr: "+stderr)
 		})
 

--- a/tests/utils/upgrade.go
+++ b/tests/utils/upgrade.go
@@ -73,7 +73,7 @@ func InstallLatestCNPGOperator(releaseTag string, env *TestingEnvironment) {
 	Eventually(func() error {
 		GinkgoWriter.Printf("installing: %s\n", mostRecentReleasePath)
 
-		_, stderr, err := RunUnchecked("kubectl apply --server-side -f " + mostRecentReleasePath)
+		_, stderr, err := RunUnchecked("kubectl apply --server-side --force-conflicts -f " + mostRecentReleasePath)
 		if err != nil {
 			GinkgoWriter.Printf("stderr: %s\n", stderr)
 		}

